ntbs-service/Services/NotificationService.cs

```diff
@@ -153,9 +154,23 @@ public async Task<Notification> GetNotificationWithNotificationSitesAsync(int? i
         public async Task UpdateSitesAsync(Notification notification, IEnumerable<NotificationSite> notificationSites) 
         {
             var currentSites = context.NotificationSite.Where(ns => ns.NotificationId == notification.NotificationId);
-            context.NotificationSite.RemoveRange(currentSites);
-            context.NotificationSite.AddRange(notificationSites);
-            await context.SaveChangesAsync();
```

Comment:
I put a bit more logic into this rather than just deleting all sites and then readding them, as before the audit logs made it look like changes were being made when they weren't.

Now it's done more properly; only adding/updating/deleting when necessary.